opengrok-indexer/src/main/java/org/opengrok/indexer/authorization/AuthorizationStack.java

```diff
@@ -277,6 +292,14 @@ protected boolean processStack(Nameable entity,
             }
         }
 
+        if (overallDecision == null && optionalFailure) {
+            return false;
+        }
+
+        if (overallDecision == null) {
+            return true;
+        }
+
```

Comment:
So the idea is that it fails the decision only when the OPTIONAL are the only plugins in the stack?

If so, the original structure may stay it was (except the OPTIONAL else-if) and here you could do

```
if(optionalFailure && getStack().stream().filter(Predicate::not(AuthorizationEntity::isOptional)).findAny().isEmpty()) {
  return optionalFailure;
}
```

Reply:
Used the original pattern according to the Unix PAM implementation (it is actually more complicated there). The stream interface makes the code arguably a bit more readable at the expense of traversing the plugins.